pkg/types/gcp/validation/platform.go

```diff
@@ -65,8 +67,34 @@ var (
 		sort.Strings(validValues)
 		return validValues
 	}()
+
+	// userLabelKeyRegex is for verifying that the label key contains only allowed characters.
+	userLabelKeyRegex = regexp.MustCompile(`^[a-z][0-9a-z_-]{1,62}$`)
+
+	// userLabelValueRegex is for verifying that the label value contains only allowed characters.
+	userLabelValueRegex = regexp.MustCompile(`^[0-9a-z_-]{1,63}$`)
+
+	// userLabelKeyPrefixRegex is for verifying that the label key does not contain restricted prefixes.
+	userLabelKeyPrefixRegex = regexp.MustCompile(`^(?i)(kubernetes\-io|openshift\-io)`)
```

Comment:
I'm surprised these are the restricted prefixes the hyphens seem weird to me. Do you recall the source of how these were determined?

Reply:
A label key can contain only hyphen and underscore special characters, unlike period character allowed in AWS, Azure. The OCP default label we add is of the form `kubernetes-io-cluster-<cluster_id>:owned`. Hence`kubernetes-io` and `openshift-io` namespaces are reserved for Openshift's usage and labels having these as prefixes are not allowed.